README.md

```diff
@@ -1,26 +1,227 @@
 # yarpc-go [![GoDoc][doc-img]][doc] [![Build Status][ci-img]][ci] [![Coverage Status][cov-img]][cov]
 
-A transport and encoding agnostic RPC framework.
+With hundreds to thousands of services communicating with RPC, transport
+protocols (like HTTP and [TChannel][]), encoding protocols (like JSON or
+Thrift), and peer choosers are the concepts that vary.
+Separating these concerns allows services to change transport protocols without
+changing call sites or request handlers, build proxies and transport protocol
+bridges, or experiment with load balancing strategies.
+YARPC is a toolkit for services and proxies.
+
+[TChannel]: https://github.com/uber/tchannel
+
+YARPC breaks RPC into interchangeable encodings, transports, and peer
+choosers.
+YARPC for Go provides reference implementations for HTTP/1.1 and TChannel
+transports, and also raw, JSON, and Thrift encodings.
+YAPRC for Go provides experimental implementations for a Redis transport and a
```

Comment:
spelling: `YARPC`.

Maybe we should add lint checks for common misspellings (like `YAPRC`)?